core/frontend/src/render/webgl/ClipStack.ts

```diff
@@ -0,0 +1,272 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module WebGL
+ */
+
+import { assert, dispose } from "@bentley/bentleyjs-core";
+import { ClipPlaneContainment, ClipVector, Point3d, Range3d, Transform } from "@bentley/geometry-core";
+import { RgbColor } from "@bentley/imodeljs-common";
+import { IModelApp } from "../../IModelApp";
+import { RenderClipVolume } from "../RenderClipVolume";
+import { FloatRgba } from "./FloatRGBA";
+import { Texture2DData, Texture2DHandle, TextureHandle } from "./Texture";
+import { ClipVolume } from "./ClipVolume";
+import { GL } from "./GL";
+import { System } from "./System";
+
+interface Clip {
+  readonly numRows: number;
+  readonly getData: (transform: Transform) => Uint8Array;
+}
+
+const emptyClipData = new Uint8Array(0);
+const emptyClip = {
+  numRows: 0,
+  getData: () => emptyClipData,
+};
+
+const scratchRangeCorners = [
+  new Point3d(), new Point3d(), new Point3d(), new Point3d(),
+  new Point3d(), new Point3d(), new Point3d(), new Point3d(),
+];
+
+function getRangeCorners(r: Range3d): Point3d[] {
+  const p = scratchRangeCorners;
+  p[0].setFromPoint3d(r.low);
+  p[1].set(r.high.x, r.low.y, r.low.z);
+  p[2].set(r.low.x, r.high.y, r.low.z);
+  p[3].set(r.high.x, r.high.y, r.low.z);
+  p[4].set(r.low.x, r.low.y, r.high.z);
+  p[5].set(r.high.x, r.low.y, r.high.z);
+  p[6].set(r.low.x, r.high.y, r.high.z);
+  p[7].setFromPoint3d(r.high);
+  return p;
```

Comment:
Couldn't you just return r.corners(scratchRangeCorners) here?

Reply:
Didn't know that was a thing.

Reply:
Done.